src/core/amount.js

```diff
@@ -614,10 +586,12 @@ function(quality, counterCurrency, counterIssuer, opts) {
     quality as stored :  5 USD          /  3000000 drops
     inverted          :  3000000 drops  /          5 USD
   */
-  var adjusted = opts.inverse ? inverse(value) : value;
-  var nativeAdjusted = adjusted;
+  const valueStr = mantissa.toString() + 'e' + offset.toString();
+  let nativeAdjusted = this._is_native ?
+    new XRPValue(valueStr) : new IOUValue(valueStr);
```

Comment:
Even if we use this hack we still run into issues with _check_limits (called from _set_value), because it's then trying to compare XRPValues with the IOUValue created in parse_quality. 
What about initially having it be an IOUValue and only at the end, when _set_value is called, turning it into an XRPValue if it's native? This will allow the >6 decimal place check in the XRPValue constructor to work appropriately